frontend/src/components/teacher/session-creation/steps/ChooseAssessment.tsx

```diff
@@ -1,23 +1,145 @@
+        assessmentFilterOptions(fetchedData.publishedTests);
+      setFilterOptions((prev) => {
+        return [
+          {
+            ...prev[0],
+            options: gradeOptions,
+          },
+          {
+            ...prev[1],
+            options: testTypeOptions,
+          },
+          {
+            ...prev[2],
+            options: countryOptions,
+          },
+          {
+            ...prev[3],
+            options: regionOptions,
+          },
+        ];
+      });
+    },
+  });
+
+  const filteredAssessments = useMemo(() => {
+    if (!data) return [];
+    if (data.publishedTests.length) {
+      setEmpty(false);
+    }
+
+    const filterProps = [grades, testTypes, countries, regions];
+    return filterAssessments(data.publishedTests, filterProps);
+  }, [data, grades, testTypes, countries, regions]);
+
+  const searchedAssessments = useMemo(() => {
+    return filterAssessmentsBySearch(filteredAssessments, search);
+  }, [filteredAssessments, search]);
+
+  const assessments = useMemo(() => {
+    return sortArray(searchedAssessments, sortProperty, sortOrder);
+  }, [searchedAssessments, sortProperty, sortOrder]);
+
   return (
-    <VStack align="left">
+    <VStack align="left" spacing="2">
       <Text color="blue.300" textAlign="left" textStyle="header4">
         Choose an Assessment
       </Text>
       <Text color="grey.300" textStyle="paragraph">
         Please enter the name of the assessment you&apos;re looking for or use
         filter/sort options to find the assessment that suits your needs.
       </Text>
+      <Box pt="6">
+        {loading ? (
+          <Center flex="1" margin="10%">
+            <LoadingState />
+          </Center>
+        ) : error || isEmpty ? (
+          <ErrorState />
+        ) : (
+          <VStack spacing={6} width="100%">
+            <HStack width="100%">
+              <SearchBar onSearch={setSearch} />
+              <SortMenu
+                initialSortOrder="ascending"
+                labels={["name", "grade", "type", "country", "region"]}
+                onSortOrder={setSortOrder}
+                onSortProperty={setSortProperty}
+                properties={[
+                  "name",
+                  "grade",
+                  "assessmentType",
+                  "curriculumCountry",
+                  "curriculumRegion",
+                ]}
+              />
+              {<FilterMenu filterProps={filterOptions} />}
+            </HStack>
+            {search && (
+              <Text color="grey.300" fontSize="16px" width="100%">
+                Showing {search.length} results for &quot;
+                {search}&quot;
+              </Text>
+            )}
+            <AssessmentsTable
+              assessments={assessments}
+              selectedTestId={testId}
+              setTestId={setTestId}
+            />
+          </VStack>
+        )}
+      </Box>
     </VStack>
   );
 };
```

Comment:
This is mainly a duplicate of the other assessments table, I'll come back to refactor when I have time (: